### PR TITLE
VB-3298 Updated Server side to store notification details in audit ta…

### PR DESCRIPTION
**Pull request by AledWEvans**
added audit log for proactive booking events

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitNotificationEventService.kt [hunk 1]
```diff
@@ -180,6 +188,18 @@ class VisitNotificationEventService(
         )
       },
     )
+
+    eventAuditRepository.saveAndFlush(
+      EventAudit(
+        actionedBy = "NOT_KNOWN",
```
[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitNotificationEventService.kt:194]: `select count(*) from event_audit ea where ea.actioned_by is null`
```
|count|
|-----|
|0    |
```
`select count(*) from event_audit ea where ea.actioned_by = 'NOT_KNOWN'`
```
|count    |
|---------|
|1,701,555|
```

[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitNotificationEventService.kt:194]: @Schema(description = "Event actioned by - user id", example = "AB12345A", required = false)
  val actionedBy: String,

It's not optional, but then require is false..

[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/service/VisitNotificationEventService.kt:194]: I think the DB should have been set to nut null after migration of old data but never was.

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Pull request merged**